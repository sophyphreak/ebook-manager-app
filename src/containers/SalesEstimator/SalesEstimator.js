import React, { Component } from 'react';

import SalesComponent from '../../components/SalesComponent/SalesComponent';

export default class SalesEstimator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bsr: '',
      sales: ''
    };
    this.onBsrChange = this.onBsrChange.bind(this);
  }

  onBsrChange(e) {
    const bsr = e.target.value;
    const sales = bsr && Math.round(10023 / bsr);
    if (!bsr || bsr.match(/^[1-9][0-9]*$/)) {
      this.setState(() => ({ bsr, sales }));
    }
  };

  render() {
    const {
      bsr,
      sales
    } = this.state;
    
    return( 
      <SalesComponent 
        bsr={bsr}
        sales={sales}
        onBsrChange={this.onBsrChange}
      />
    );
  };
};