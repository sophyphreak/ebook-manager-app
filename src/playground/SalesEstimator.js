import React, { Component } from 'react';
import {
  Card,
  CardBody,
  CardHeader,
  Col,
  FormGroup,
  Input,
  Label,
  Row,
} from 'reactstrap';
import BasicInput from '../components/FormElements/MainElements/BasicInput';

export default class SalesEstimator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      bsr: '',
      sales: ''
    };
    this.onBsrChange = this.onBsrChange.bind(this);
  }

  onBsrChange(e) {
    const bsr = e.target.value;
    const sales = bsr && Math.round(10023 / bsr);
    if (!bsr || bsr.match(/^[1-9][0-9]*$/)) {
      this.setState(() => ({ bsr, sales }));
    }
  };

  render() {
    return (
      <Row>
        <Col xs="12" sm={{ size: 6, offset: 3 }}>
          <Card>
            <CardHeader>
              <h4><strong>Sales Estimator</strong></h4>
            </CardHeader>
            <CardBody>
              <BasicInput 
                label="Please enter the BSR"
                onChange={this.onBsrChange} 
                value={this.state.bsr} 
              />
              <br />
              {
                this.state.bsr && 
                <FormGroup className="animated fadeIn">
                  <Label>Estimated sales</Label>
                  <br />
                  <Label><b>{this.state.sales} units per month</b></Label>
                </FormGroup>
              }
            </CardBody>
          </Card>
        </Col>
      </Row>
    );
  };
};