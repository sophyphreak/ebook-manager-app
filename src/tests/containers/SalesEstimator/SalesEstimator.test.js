import React from 'react';
import { shallow } from 'enzyme';
import SalesEstimator from '../../../containers/SalesEstimator/SalesEstimator';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<SalesEstimator />);
});

test('should render SalesEstimator correctly', () => {
  expect(wrapper).toMatchSnapshot();
});