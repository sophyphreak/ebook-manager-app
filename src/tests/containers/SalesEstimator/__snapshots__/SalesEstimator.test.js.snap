// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SalesEstimator correctly 1`] = `
<SalesComponent
  bsr=""
  onBsrChange={[Function]}
  sales=""
/>
`;
